homeassistant/components/bring/todo.py

```diff
@@ -0,0 +1,186 @@
+
+from .const import DOMAIN
+from .coordinator import BringDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=90)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor from a config entry created in the integrations UI."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    unique_id = config_entry.unique_id
+
+    if TYPE_CHECKING:
+        assert unique_id
+
+    for _list in coordinator.data:
+        async_add_entities(
+            [
+                BringTodoListEntity(
+                    coordinator,
+                    name=f"Bring! {_list['name']}",
+                    unique_id=f"{unique_id} {_list['listUuid']}",
+                )
+            ],
+        )
+
+
+class BringTodoListEntity(
+    CoordinatorEntity[BringDataUpdateCoordinator], TodoListEntity
+):
+    """A To-do List representation of the Bring! Shopping List."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+        | TodoListEntityFeature.SET_DESCRIPTION_ON_ITEM
+    )
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: BringDataUpdateCoordinator,
+        name: str,
+        unique_id: str,
+    ) -> None:
+        """Initialize BringTodoListEntity."""
+        super().__init__(coordinator)
+        self._attr_icon = "mdi:cart"
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+
+    async def async_added_to_hass(self) -> None:
+        """Populate the todo list at start."""
+        self._populate_list()
+        await super().async_added_to_hass()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle the state update and prepare the todo list items."""
+        self._populate_list()
+        return super()._handle_coordinator_update()
+
+    @property
+    def list_(self) -> BringList | None:
+        """Return the bring list."""
+        return next(
+            (
+                _list
+                for _list in self.coordinator.data
+                if _list["listUuid"] == self.unique_id
+            ),
+            None,
+        )
+
+    def _populate_list(self) -> None:
+        """Populate the todo list using the coordinator data."""
+
+        _list = self.list_
+        if _list is not None:
+            self._attr_todo_items = [
+                TodoItem(
+                    uid=item["name"],
+                    summary=item["name"],
+                    description=item["specification"] or "",
+                    status=TodoItemStatus.NEEDS_ACTION,
+                )
+                for item in _list["items"]
+            ]
+        else:
+            self._attr_todo_items = None
```

Comment:
Since this function is this little, have you considered just using a property? It saves you the async_added_to_hass and _handle_coordinator_update.

Reply:
@joostlek I tried using a property, but this does not get refreshed when I call "update" on the coordinator:

```
    @property
    def todo_items(self) -> list[TodoItem]:
        """Return the todo items."""
        return [
            TodoItem(
                uid=item["name"],
                summary=item["name"],
                description=item["specification"] or "",
                status=TodoItemStatus.NEEDS_ACTION,
            )
            for item in self.bringList["items"]
        ]
```

This function is run only once at the beginning. How can I trigger it again?